app/src/main/java/com/example/dispatcher/common/base/SplashActivity.kt

```diff
@@ -1,21 +1,33 @@
 package com.example.dispatcher.common.base
 
+import android.content.Intent
 import android.os.Bundle
 import androidx.activity.ComponentActivity
-import android.content.Intent
 import android.os.Handler
 import com.example.dispatcher.R
+import com.google.firebase.auth.FirebaseAuth
 
 class SplashActivity : ComponentActivity() {
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_splash)
 
         Handler().postDelayed({
-            // Start MainActivity
+            checkUserStatus()
+        }, 3000)
+    }
+
+    private fun checkUserStatus() {
+        val currentUser = FirebaseAuth.getInstance().currentUser
+
+        if (currentUser != null) {
             val intent = Intent(this@SplashActivity, MainActivity::class.java)
```

Comment:
this is duplicated code, the if statement can decide the activity. then you can create the intent with it.